### PR TITLE
Feat/1609 date format dynamic expression function

### LINKED ISSUE #1609: Need a way to format datetime fields in the frontend
### Description

Currently, there is no other way to display a date in the specified users locale other than rendering a datepicker and using `displayValue`, expression on that component. 

I want to be able to use a function like `formatDateLocale` in an expression to be able to show dates that are now connected to a component on the correct format. 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by mikaelrss**
## Description

Adds a new dynamic expression function `formatDate`. The function takes two parameters:
- date - The ISO-string that should be formatted.
- format - A format in which the date should be formatted (optional). If this argument is omitted, the function will use the selected language to find the correct locale, and calculate an appropriate format based on this. 

## Related Issue(s)

- closes #1609 

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [x] Unit test(s) have been added/updated
  - [ ] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [x] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [x] Has been added/updated
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [x] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Comment by mikaelrss** [comment]
The Percy tests that are failing are because of new options added to the frontend-test application for testing dynamic expressions in list components based on repeating groups. They are not introduced because of these changes.

**Review by bjosttveit — APPROVED** [review]
Great!

ps: remember to update the table of functions in docs

**Comment by mikaelrss** [comment]
Yes, I was planning on adding it to v3 together with the `sortOrder` functionality on selectable components :partying_face:

**Comment by bjosttveit** [comment]
Nice! In that case, remember to update the table of functions in the docs 🙌

**Comment by mikaelrss** [comment]
Will do!

**Review by olemartinorg — COMMENTED** [review]
Maybe we should have some shared tests for this as well? 🤔 In case it will be implemented on the backend as well.

**Comment by olemartinorg** [comment]
@mikaelrss Have a look at the readme: https://github.com/Altinn/app-frontend-react/tree/main/src/features/expressions/shared-tests

The shared tests are unit-tests structured in JSON files that define all the input needed, along with the expected expression result. We have tests-runners implemented both in frontend and in backend (`app-lib-dotnet`), as there are two expression engine implementations. I don't see the need for a backend implementation for this function right now, but having at least one shared test case would make it pretty clear where these two implementations diverge - and would nudge us to remember adding a backend implementation at some point, should we need it.

**Review by olemartinorg — APPROVED** [review]

**Review by olemartinorg — APPROVED** [review]
What a glorious piece of code! 🤩 Loving it! Especially the fantastically complex and thorough unit-test.. 😍 There was one eslint complaint about nested template strings, but otherwise this looks good!

**Comment by olemartinorg** [comment]
Also, should we add to existing issues or create a new one to migrate over to this from moment, etc? 🤔

File: package.json [hunk 1]
```diff
@@ -13,9 +13,9 @@
     "build": "yarn gen && yarn copy-schemas && cross-env NODE_ENV=production webpack --config webpack.config.production.js --progress",
     "docker": "yarn gen && yarn copy-schemas && cross-env NODE_ENV=development webpack --config webpack.config.development.js --progress && yarn serve 8080",
     "serve": "http-server dist --cors='*' -p",
-    "test": "yarn gen && jest",
-    "test:watch": "yarn gen && jest --watch",
-    "test:watchall": "yarn gen && jest --watchAll",
+    "test": "yarn gen && TZ=UTC jest",
+    "test:watch": "yarn gen && TZ=UTC jest --watch",
+    "test:watchall": "yarn gen && TZ=UTC jest --watchAll",
```
[the remark this answers is not in the document]
Comment (mikaelrss) [reply package.json:18]: Aha, I can do some extra checking to find out if it would work on windows :sweat_smile:

**Comment by mikaelrss** [comment]
> What a glorious piece of code! 🤩 Loving it! Especially the fantastically complex and thorough unit-test.. 😍 There was one eslint complaint about nested template strings, but otherwise this looks good!

:partying_face: Do you want me to refactor it to not use nested template strings, or should I disable the lint-rule for that line?

> Also, should we add to existing issues or create a new one to migrate over to this from moment, etc? 🤔

Maybe we could keep the original issue(s), and tag it with v4? I can reference this issue in the others.

**Comment by olemartinorg** [comment]
> Do you want me to refactor it to not use nested template strings, or should I disable the lint-rule for that line?

Do what you prefer. I often refactor instead of using `eslint-disable` for such little things, and in some ways I agree with the rule that nested template strings quickly becomes hard to read. However, I didn't read this one too closely, so if you disagree with the readability it might work just as well just skipping the rule.

> Maybe we could keep the original issue(s), and tag it with v4? I can reference this issue in the others.

Yup, go ahead and reference this in the other issues - but I'm not sure we need the v4 label. The reason the label is somewhat gold colored is because it should be used for all issues we need to complete in order to ship v4 (aka 'go gold'), but I'm not sure replacing moment (and thus the date picker?) is a hard requirement for v4.

**Pull request merged**